lib/util/git/index.ts

```diff
@@ -503,11 +520,9 @@ export async function checkoutBranch(branchName: string): Promise<CommitSha> {
   logger.debug(`Setting current branch to ${branchName}`);
   await syncGit();
   try {
-    config.currentBranch = branchName;
-    config.currentBranchSha = (
-      await git.raw(['rev-parse', 'origin/' + branchName])
-    ).trim();
     await gitRetry(() => git.checkout(['-f', branchName, '--']));
+    config.currentBranch = branchName;
```

Comment:
it's necessary to avoid branch names with origin prefix and it's a bug (checkout can fail but  currentBranch was changed and not restored)

i can extract it to a separate pr too.

Reply:
please do so